src/pyTRLCConverter/markdown_converter.py

```diff
@@ -120,56 +240,235 @@ def convert_record_object(self, record: Record_Object, level: int) -> Ret:
         Returns:
             Ret: Status
         """
-        record_attributes = record.to_python_dict()
+        assert self._fd is not None
 
-        markdown_heading = self.markdown_create_heading(record.name, level + 1)
-        self._fd.write(markdown_heading)
+        self._write_empty_line_on_demand()
+        return self._convert_record_object(record, level, None)
 
-        column_titles = ["Attribute Name", "Attribute Value"]
-        markdown_table_head = self.markdown_create_table_head(column_titles)
-        self._fd.write(markdown_table_head)
+    def finish(self):
+        # lobster-trace: SwRequirements.sw_req_markdown_single_doc_mode
+        """Finish the conversion process.
+        """
 
-        for key, value in record_attributes.items():
-            if value is None:
-                value = "N/A"
+        # Single document mode?
+        if self._args.single_document is True:
+            assert self._fd is not None
+            self._fd.close()
+            self._fd = None
 
-            elif isinstance(value, list):
-                value = ", ".join(value)
+        return Ret.OK
 
-            markdown_table_row = self.markdown_append_table_row([key, value])
-            self._fd.write(markdown_table_row)
+    def _write_empty_line_on_demand(self) -> None:
+        # lobster-trace: SwRequirements.sw_req_markdown
+        """Write an empty line if necessary.
+        """
+        if self._empty_line_required is False:
+            self._empty_line_required = True
+        else:
+            self._fd.write("\n")
 
-        self._fd.write("\n")
+    def _get_markdown_heading_level(self, level: int) -> int:
+        # lobster-trace: SwRequirements.sw_req_markdown_record
+        """Get the Markdown heading level from the TRLC object level.
+            Its mandatory to use this method to calculate the Markdown heading level.
+            Otherwise in single document mode the top level heading will be wrong.
 
-        return Ret.OK
+        Args:
+            level (int): The TRLC object level.
+        
+        Returns:
+            int: Markdown heading level
+        """
+        return self._base_level + level
+
+    def _file_name_trlc_to_md(self, file_name_trlc: str) -> str:
+        # lobster-trace: SwRequirements.sw_req_markdown_multiple_doc_mode
+        """Convert a TRLC file name to a Markdown file name.
+
+        Args:
+            file_name_trlc (str): TRLC file name
+        
+        Returns:
+            str: Markdown file name
+        """
+        file_name = os.path.basename(file_name_trlc)
+        file_name = os.path.splitext(file_name)[0] + ".md"
+
+        return file_name
 
     def _generate_out_file(self, file_name: str) -> Ret:
         # lobster-trace: SwRequirements.sw_req_markdown_out_folder
         """Generate the output file.
 
         Args:
-            file_name (str): File name of the input file.
+            file_name (str): File name which to use, without path.
```

Comment:
"which to use"  is not adding information.
just "The output file name without path."

Reply:
ok